### ISSUE TITLE
[a11y]: Link text is repeated in an image 'alt' value within the same link

### ISSUE THREAD

**Issue by huangkevin-apr** [carbon-design-system/carbon-website#4777]
## Environment

> Operating system
Windows 11

> Browser
Edge

> Automated testing tool and ruleset
[IBM Equal Access Accessibility Checker](https://github.com/IBMa/equal-access/tree/main-4.x) (ruleset ["4.0.9"](https://github.com/IBMa/equal-access/blob/main-4.x/rule-server/src/static/archives.json))

> Assistive technology used to verify
Screen Reader

## Detailed description

> What version of the Carbon Design System are you using?
Current Version

> What did you expect to happen?
When navigating interactive card links using a screen reader, the accessible name of each link should be announced once, without redundant or repeated content.

> What happened instead?
For card-style links that contain both visible text (e.g., a heading) and an image with an ```alt``` attribute inside the same ```<a>``` element, the screen reader announces the same text twice.
This occurs because the image ```alt``` value repeats the link text, resulting in redundant accessible name computation.

<img width="1858" height="536" alt="Image" src="https://github.com/user-attachments/assets/8c976e06-97ef-4fff-9cd6-09bdbc8b1b02" />

> What WCAG 2.1 checkpoint does the issue violate?
- [SC 1.1.1](https://w3.org/TR/WCAG22#non-text-content):Non-text Content (Level A)
- [SC 2.4.4](https://w3.org/TR/WCAG22#link-purpose-in-context): Link Purpose (In Context) (Level A)

## Steps to reproduce the issue

1. Open the Carbon Design System website (https://carbondesignsystem.com/).
2. Run the [IBM A11Y Checker V 4.0.9](https://www.ibm.com/able/toolkit/tools/).
3. View the Violation Report.

## Additional information

- Screenshots of 3 Violations

<img width="2560" height="921" alt="Image" src="https://github.com/user-attachments/assets/0f633efe-b181-474b-8c51-171330d526fb" />

<img width="2560" height="921" alt="Image" src="https://github.com/user-attachments/assets/c46dbac0-1bd6-4fe8-9d54-bf7f6fe75373" />

<img width="2560" height="922" alt="Image" src="https://github.com/user-attachments/assets/98add2ce-f5c1-4b10-bb46-16b3503d0c9c" />
